packages/mimir/src/rules/no-return-await.ts

```diff
@@ -13,11 +13,15 @@ export class Rule extends AbstractRule {
             const {node} = getWrappedNodeAtPosition(wrappedAst || (wrappedAst = this.context.getWrappedAst()), re.lastIndex - 1)!;
             if (isAwaitExpression(node) && re.lastIndex === node.expression.pos && isUnnecessaryAwait(node)) {
                 const keywordStart = node.expression.pos - 'await'.length;
+                const replacement =
+                    (node.expression.kind === ts.SyntaxKind.ObjectLiteralExpression)
```

Comment:
this condition also needs to check if `node.parent` is an ArrowFunction

and add the following test case to ensure it works as expected:
```ts
// inside an async function
return await {foo: 1}; // doesn't need parens
```